File: ui_ux/backend/models/Teacher.js
const mongoose = require("mongoose");
const bcrypt = require("bcrypt");

const Schema = mongoose.Schema;

const teacherSchema = new Schema({
  fullName: {
    type: String,
    required: true,
    min: 2,
    max: 50,
  },
  gender: {
    type: String,
    required: true,
    default: "Male",
  },
  experience: {
    type: String,
    required: true,
  },
  location: {
    type: String,
    required: true,
    max: 50,
  },
  email: {
    type: String,
    required: true,
    max: 20,
    unique: true,
  },
  phoneNumber: {
    type: String,
    required: true,
    min: 6,
    max: 12,
  },
  // password: {
  //   type: String,
  //   required: true,
  //   min: 5,
  // },
  occupation: {
    type: String,
    required: true,
  },

  institution: {
    type: String,
    required: true,
    // min:2,
    max: 50,
  },
  subject: {
    type: String,
    required: false,
  },
  picturePath: {
    type: String,
    default: "",
    required: false,
  },
  teachingSubject: {
    type: Array,
    required: false,
    default: [],
  },
  rating: {
    type: [Number], 
    default: [],
  },
  minSalary: {
    type: String,
    required: true,
  },
  maxSalary: {
    type: String,
    required: true,
  },
  studentId: {
    type: Array,
    //  required: true,
    default: [],
  },
});

module.exports = mongoose.model("Teacher", teacherSchema);

File: ui_ux/lib/models/teacher.dart
import 'dart:convert';

class Teacher {
  final String id;
  final String fullName;
  final String gender;
  final String experience;
  final String location;
  final String email;
  final String phoneNumber;
  final String occupation;
  final String institution;
  final String subject;
  String? picturePath;
  final String teachingSubject;
  final String minSalary;

  var rating;

  final String maxSalary;

  Teacher(
      {required this.id,
      required this.fullName,
      required this.gender,
      required this.experience,
      required this.location,
      required this.email,
      required this.phoneNumber,
      required this.occupation,
      required this.institution,
      required this.subject,
      this.picturePath,
      required this.teachingSubject,
      required this.minSalary,
      required this.maxSalary,
      this.rating});

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      "fullName": fullName,
      "gender": gender,
      "experience": experience,
      "location": location,
      "email": email,
      "phoneNumber": phoneNumber,
      "occupation": occupation,
      "institution": institution,
      "subject": subject,
      "picturePath": picturePath,
      "teachingSubject": teachingSubject,
      "minSalary": minSalary,
      "maxSalary": maxSalary,
    };
  }

  String toJson() => json.encode(toMap());
}
